Content.Server/Dragon/DragonSystem.cs

```diff
@@ -0,0 +1,195 @@
+            var halfedIchorInjection = new Solution(dragoncomp.DevourChem, dragoncomp.DevourHealRate / 2);
+
+            //Check if the target is valid. The effects should be possible to accomplish on either a wall or a body.
+            //Eating bodies is instant, the wall requires a doAfter.
+
+            //NOTE: I honestly don't know much on how one detects valid eating targets, so right now I am using a body component to tell them apart
+            //That way dragons can't devour guardians and other dragons. Yet.
+
+            if (EntityManager.TryGetComponent(target, out MobStateComponent targetstate))
+            {
+                if (targetstate.CurrentState != null)
+                {
+                    //You can only devour dead or crit targets
+                    if (targetstate.CurrentState.IsIncapacitated())
+                    {
+                        if (EntityManager.TryGetComponent(dragonuid, out DamageableComponent dragonhealth))
+                        {
+                            //Humanoid devours allow dragon to get eggs, corpses included
+                            if (EntityManager.TryGetComponent(target, out HumanoidAppearanceComponent humanoid))
+                            {
+                                dragoncomp.EggsLeft++;
+                                //inject the healing chemical into the system. Yes the dragon just fucking drinks his own blood.
+                                _bloodstreamSystem.TryAddToChemicals(dragonuid, ichorInjection);
+                                //Sends the human entity into the stomach so it can be revived later. Withold further comments.
+                                dragoncomp.DragonStomach.Insert(target);
+                                SoundSystem.Play(Filter.Pvs(dragonuid), "/Audio/Effects/sound_magic_demon_consume.ogg");
+
+                            }
+                            //Non-humanoid mobs can only heal dragon for half the normal amount
+                            else
+                            {
+                                //heal HALF the damage
+                                _bloodstreamSystem.TryAddToChemicals(dragonuid, halfedIchorInjection);
+                                //Sends the non-human entity into the stomach
+                                //NOTE: I am a bit conflicted on this, and only really adding this because force-delete is bad, is this needed?
+                                dragoncomp.DragonStomach.Insert(target);
+                                SoundSystem.Play(Filter.Pvs(dragonuid), "/Audio/Effects/sound_magic_demon_consume.ogg");
+                            }
+                        }
+                        else return;
+
+                    }
+                    else _popupSystem.PopupEntity(Loc.GetString("devour-action-popup-message-fail-target-alive"), dragonuid, Filter.Entities(dragonuid));
+                    return;
+                }
+                else return;
+            }
```

Comment:
WAY too indented you need to invert the control flow here in some places (`if (!comp) return;` rather than `(if comp) { } else { return; }`)

Reply:
Also please don't inline else statements.